filament/src/PostProcessManager.h

```diff
@@ -393,7 +384,60 @@ class PostProcessManager {
             ColorGradingConfig const& colorGradingConfig, VignetteOptions const& vignetteOptions,
             uint32_t const width, uint32_t const height) noexcept;
 
+    void resetForRender();
+
 private:
+    // This class manages a cache of material instances mapped by materials.  This allows us to
+    // re-use instances across frames. Note that also it's possible for the same frame to want to
+    // use the same instance multiple times (for example for bilateral blurring or upsampling), but
+    // we allocate separate instances for that case so that UBOs are not shared across render
+    // passes. This allows for bubble-reduction for vulkan (and possibly other backends).
+    class MaterialInstanceManager {
```

Comment:
Could we move this into its own file? And just keep the helper for PostProcessMaterial into PostProcessManager (so that MaterialInstanceManager doesn't need to depend on PostProcessMateral).
I think this class is 99% generic, we might as well make it 100%.